packages/interactivity-router/src/index.ts

```diff
@@ -321,9 +327,10 @@ window.addEventListener( 'popstate', async () => {
 	const pagePath = getPagePath( window.location.href ); // Remove hash.
 	const page = pages.has( pagePath ) && ( await pages.get( pagePath ) );
 	if ( page ) {
-		renderPage( page );
-		// Update the URL in the state.
-		state.url = window.location.href;
+		batch( () => {
+			state.url = window.location.href;
+			renderPage( page );
```

Comment:
I see you're wrapping `renderPage` calls with `batch()`. Should we remove the `batch()` used internally by this function? 🤔

Reply:
Nested `batch` calls are not harmful, but it's true that they don't have any benefit either.

My idea for leaving it like that was that it makes sense for `renderPage` to do a `batch` internally, but apart from that, I wanted to minimize the re-render caused by the change of `state.url` and have that happen at the same time. I thought about moving `state.url` inside `renderPage`, but it felt weird to make `renderPage` had a side effect on the store.

In the end, since I'm not entirely sure what the best way is to join all the invalidations to minimize re-renders, I opted to leave the calls to `batch` explicit, like this. We can review it in the future when we start investigating a bit how the Interactivity API handles re-renders and if there is anything that can be improved.